docs: define the update.conf (#125)
kinvolk/flatcar-docs: docs/clusters/management/update-conf.md

---
content_type: reference
title: update.conf
description: >
  Definitions for Flatcar update configuration.
weight: 100
---

Flatcar Container Linux uses [`update_engine`][update_engine] to check and fetch new updates from the [Nebraska Update Service](https://github.com/kinvolk/nebraska).

## Location

> The client-side configuration these updates is stored in `/etc/flatcar/update.conf`.

Reviewer: ```suggestion
The client-side configuration of these updates is stored in `/etc/flatcar/update.conf`.
```

(there is a legacy symlink of `/etc/coreos -> /etc/flatcar` for compatibility with third-party integrations).
This file is in the user writable partition by default.

Next order of client-side configurations that are checked are:

* `/usr/share/flatcar/update.conf`
  * Generated at build time of the image/payload build
  * will typically contain:
    * `SERVER=`
    * `GROUP=`
* `/usr/share/flatcar/release`
  * Generated at build time of the image/payload build
  * will typically contain:
    * `FLATCAR_RELEASE_VERSION=`
    * `FLATCAR_RELEASE_BOARD=`
    * `FLATCAR_RELEASE_APPID=`

## Fields

> Default installs of Flatcar will likely only have `GROUP=` set.

Reviewer 2: This sentence doesn't sound right. All have to be set but the defaults can be used.

I would reword it as:
```suggestion
Default installs of Flatcar will likely not need custom settings, and an empty `/etc/flatcar/update.conf` is sufficient.
```

* `GROUP`
  * The channel/group this host will pull updates from

>   * public channels will be: `stable`, `beta`, `alpha`

Reviewer: What about `edge`?

Author: we'll be renaming (or hopefully removing) edge. These three are the long established ones.

  * otherwise these are UUIDs
* `SERVER`
  * The update server to reach for updates
  * default community server is: https://public.update.flatcar-linux.net/v1/update/
* `FLATCAR_RELEASE_VERSION`

>   * The current version of this machine's version

Reviewer 2: ```suggestion
  * The current version of this machine
```
